airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/retriever.py

```diff
@@ -36,3 +37,8 @@ def get_state(self) -> MutableMapping[str, Any]:
          State should try to be as small as possible but at the same time descriptive enough to restore
          syncing process from the point where it stopped.
         """
+
+    @state.setter
```

Comment:
This is an interesting pattern. I was under the impression that python didn't really support overloading, but looks like this is a thing